src/Build/Logging/BinaryLogger/Postprocessing/IBuildFileReader.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Build.Logging;
+
+public interface IBuildFileReader
+{
+    /// <summary>
+    /// An event that allows the caller to be notified when an embedded file is encountered in the binary log.
+    /// Subscribing to this event obligates the subscriber to read the file content and set the resulting content
+    ///  via <see cref="ArchiveFileEventArgs.SetResult(string,System.IO.Stream)"/> or <see cref="ArchiveFileEventArgs.SetResult(string,string)"/>.
+    /// When subscriber is OK with greedy reading entire content of the file, it can simplify subscribing to this event,
+    /// by using handler with same signature as handler for <see cref="IBuildEventStringsReader.StringReadDone"/> and wrapping it via
+    /// <see cref="ArchiveFileEventArgsExtensions.ToArchiveFileHandler"/> extension.
```

Comment:
I don't understand this. How can the interface be the same for archive-of-files vs strings?

Reply:
Clarified the comment.
It applies only when reader is interrested in strings only (e.g. to redact sensitive data) - without need to distinguish what is a path and what is a content.